e2e/cypress/support/commands.js

```diff
@@ -265,3 +265,19 @@ Cypress.Commands.add('waitForQc', (timeout, numQcSteps = 7) => {
   log.snapshot('qc-completed');
   log.end();
 });
+
+Cypress.Commands.add('cleanUpProjectIfNecessary', (projectName) => {
+  const log = Cypress.log({
+    displayName: 'Cleaning up previous project',
+    message: ['clean project'],
+  });
+
+  const projectExists = Cypress.$(`span:contains(${projectName})`).length;
```

Comment:
I think the selector here should include `data-test-project-card` to make it more clear and robust. Maybe you can just do `data-test-project-card.:contains(${projectName})`, otherwise you might need to do  `data-test-project-card > span.:contains(${projectName})`. Essentially, I'd prefer to remove the `span` if possible, because it couples the HTML tag to the test. The test then would break if we ever change the HTML element as opposed to just using the `data-test-project-card`  which should never change.